chains/evm/contracts/ocr/SignatureQuorumVerifier.sol

```diff
@@ -2,150 +2,169 @@
-      // Safe from ECDSA malleability here since we check for duplicate signers.
-      address signer = ecrecover(hashedReport, 27, rs[i], ss[i]);
-
-      // Check that the signer is registered as an oracle.
-      if (!s_signers.contains(signer)) revert UnauthorizedSigner();
+      // We use ECDSA malleability to only have signatures with a `v` value of 27.
+      address signer = ecrecover(reportHash, 27, proofs.rs[i], proofs.ss[i]);
+      // Check that the signer is registered.
+      if (!config.signers.contains(signer)) revert UnauthorizedSigner();
       // This requires ordered signatures to check for duplicates. This also disallows the zero address.
       if (uint160(signer) <= lastSigner) revert NonUniqueSignatures();
+      lastSigner = uint160(signer);
     }
   }
 
-  // ================================================================
-  // │                           Config                             │
-  // ================================================================
+  /// @notice Returns all active config digests.
+  function getActiveConfigDigests() external view returns (bytes32[] memory) {
+    return s_activeConfigDigests.values();
+  }
 
-  /// @notice Sets offchain reporting protocol configuration incl. participating oracles for a single OCR plugin type.
-  /// @param ocrConfigArgs OCR config update args.
-  function setSignatureConfig(
-    SignatureConfigArgs memory ocrConfigArgs
-  ) external onlyOwner {
-    if (ocrConfigArgs.F == 0) revert InvalidConfig();
+  /// @notice Returns all active config digests and their corresponding signer sets, and F values.
+  function getAllActiveConfigs() external view returns (SignatureConfigArgs[] memory) {
+    bytes32[] memory configDigests = s_activeConfigDigests.values();
+    SignatureConfigArgs[] memory configs = new SignatureConfigArgs[](configDigests.length);
+
+    for (uint256 i = 0; i < configDigests.length; ++i) {
+      configs[i] = SignatureConfigArgs({
+        configDigest: configDigests[i],
+        F: s_signatureConfig[configDigests[i]].F,
+        signers: s_signatureConfig[configDigests[i]].signers.values()
+      });
+    }
+    return configs;
+  }
 
-    address[] memory newSigners = ocrConfigArgs.signers;
+  /// @notice Sets a new signature configuration for a given configDigest.
+  /// @param signatureConfigs The configuration to set, containing the configDigest, F value, and signers.
+  /// @dev Reverts if the configDigest already exists, this function cannot override an existing configuration.
+  function setSignatureConfigs(
+    SignatureConfigArgs[] calldata signatureConfigs
+  ) external onlyOwner {
+    for (uint256 i = 0; i < signatureConfigs.length; ++i) {
+      SignatureConfigArgs memory signatureConfig = signatureConfigs[i];
 
-    _clearAllSigners();
+      if (
+        signatureConfig.F == 0 || signatureConfig.F > signatureConfig.signers.length
+          || signatureConfig.configDigest == ""
```

Comment:
Just want to call out that we should be able to reliably generate the configDigest offchain so we can avoid multi-proposal operational workflows (i.e. we can set a config on the home chain and set the digest here in the same proposal without any issues)